A39-xds-http-filters.md

```diff
@@ -0,0 +1,152 @@
+A39: xDS HTTP Filter Support
+----
+* Author(s): Mark D. Roth (markdroth)
+* Approver: ejona86, dfawley
+* Status: In Review
+* Implemented in: 
+* Last updated: 2021-02-03
+* Discussion at: https://groups.google.com/g/grpc-io/c/M-l8k2v5snY
+
+## Abstract
+
+The purpose of this proposal is to describe how gRPC will support HTTP
+filters configured via the xDS API.
+
+## Background
+
+gRPC currently supports obtaining configuration via the
+[xDS API](https://github.com/envoyproxy/data-plane-api), as per
+[gRFC A27](https://github.com/grpc/proposal/blob/master/A27-xds-global-load-balancing.md).
+
+This proposal details how gRPC will add support for xDS functionality that is
+configured via [HTTP
+filters](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/http_filters).  Initially, we will not provide any sort of public
+API that third parties can use to implement their own filters, although
+that may come later.  For now, we are focusing on creating a structure
+that will enable us to implement functionality that, in Envoy, is
+provided by its out-of-the-box filters (e.g., fault injection or
+authorization), which we also want to be provided out-of-the-box in gRPC.
+
+### Related Proposals: 
+
+- [A27: xDS-Based Global Load Balancing](https://github.com/grpc/proposal/blob/master/A27-xds-global-load-balancing.md)
+- [A30: xDS v3 Support](https://github.com/grpc/proposal/blob/master/A30-xds-v3.md)
+- [A31: gRPC xDS Timeout Support and Config Selector Design](https://github.com/grpc/proposal/blob/master/A31-xds-timeout-support-and-config-selector.md)
+
+## Proposal
+
+In the gRPC client channel, the xDS HTTP filters will run in between
```

Comment:
What about on the gRPC server side? Should we say something about whether we are/will be planning to support HTTP filters on the server side e.g. [rbac_filter](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/rbac_filter) which is a server side one?

Reply:
We've decided that we will support HTTP filters on the server side.  I've updated this gRFC accordingly, and @ejona86 will update gRFC A36 to (a) cover L4 filters, (b) refer to this gRFC for discussion of HTTP filters, and (c) cover handling of RouteConfiguration on the server side.